wgpu-core/src/swap_chain.rs

```diff
@@ -245,14 +243,21 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
             }
         }
 
-        Ok(())
+        Ok(match result {
+            Ok(None) => (SwapChainStatus::Good),
+            Ok(Some(_)) => (SwapChainStatus::Suboptimal),
+            Err(err) => match err {
+                hal::window::PresentError::OutOfMemory(_) => SwapChainStatus::OutOfMemory,
```

Comment:
some of these are not really recoverable by wgpu-rs users
`OutOfMemory` and `DeviceLost` are similar to other places where this can happen, and we don't currently provide any way to handle these gracefully.
So maybe we should really keep the statuses for those things that belong to recoverable errors? I think `OutOfData` and `SurfaceLost` are only those.

Reply:
Shall I remove those variants from SwapChainStatus for both swap_chain_get_current_texture_view and swap_chain_present in this PR?

Reply:
I went ahead and did that.